packages/typescript/src/create-package.ts

```diff
@@ -54,28 +58,81 @@ function createSymbols(
     }
 
     for (const exportedName of symbols.named ?? []) {
-      const key = keys[exportedName];
-      const sym = createTSSymbol({
-        name: exportedName,
-        scope: moduleScope,
-        refkey: key,
-        export: true,
-        default: false,
-      });
-      moduleScope.exportedSymbols.set(key, sym);
+      if (typeof exportedName === "object") {
+        const { name, staticMembers } = exportedName;
+        const key = keys[name];
+        const sym = createTSSymbol({
+          name,
+          scope: moduleScope,
+          refkey: key,
+          export: true,
+          default: false,
+        });
+        moduleScope.exportedSymbols.set(key, sym);
+
+        for (const staticMember of staticMembers ?? []) {
+          const staticKey = refkey(`${key}.${staticMember}`);
+          const staticSym = createTSSymbol({
+            name: staticMember,
+            scope: moduleScope,
+            refkey: staticKey,
+            export: true,
+            default: false,
+          });
+          moduleScope.exportedSymbols.set(staticKey, staticSym);
+        }
+      } else {
+        const key = keys[exportedName];
+        const sym = createTSSymbol({
+          name: exportedName,
+          scope: moduleScope,
+          refkey: key,
+          export: true,
+          default: false,
+        });
+        moduleScope.exportedSymbols.set(key, sym);
+      }
     }
   }
 }
 
+// TODO: simplify this crazy type?
+// Helper type to extract named exports as strings
+type NamedStringExports<T extends ModuleSymbolsDescriptor> = {
```

Comment:
Madness 😄

Reply:
I mean, if it works....

Reply:
Fair enough. I tried to refactor it a bit where possible